src/main/java/uk/gov/companieshouse/api/strikeoffobjections/service/impl/ObjectionService.java

```diff
@@ -78,28 +80,45 @@
     private ActionCodeValidator actionCodeValidator;
 
     @Override
-    public Objection createObjection(String requestId, String companyNumber, String ericUserId, String ericUserDetails) {
+    public Objection createObjection(String requestId,
+                                     String companyNumber,
+                                     ObjectionUserDetails userDetails) {
+
         Map<String, Object> logMap = buildLogMap(companyNumber, null, null);
         logger.infoContext(requestId, "Creating objection", logMap);
 
         final Long actionCode = getActionCode(companyNumber, requestId);
-
         final ObjectionStatus objectionStatus = getObjectionStatusForCreate(actionCode, companyNumber, requestId);
 
-        final String userEmailAddress = ericHeaderParser.getEmailAddress(ericUserDetails);
-
         Objection entity = new Objection.Builder()
                 .withCompanyNumber(companyNumber)
                 .withCreatedOn(dateTimeSupplier.get())
-                .withCreatedBy(new CreatedBy(ericUserId, userEmailAddress))
+                .withCreatedBy(userDetails.getCreatedby())
                 .withHttpRequestId(requestId)
                 .withActionCode(actionCode)
                 .withStatus(objectionStatus)
+                .withFullName(userDetails.getFullName())
```

Comment:
So if 'fullName' and 'shareIdentity' lived alongside 'email' in the created_by bucket, these lines wouldn't be needed.

Reply:
Removed